Runtime/ui/txt/paragraph.cs

```diff
@@ -298,8 +298,15 @@ public struct LineRange {
 
             for (int i = 0; i < this._paintRecordsCount; i++) {
                 var paintRecord = this._paintRecords[i];
-                this._textPaint.color = paintRecord.style.color;
+                
+                if (paintRecord.style.foreground != null) {
+                    this._textPaint = paintRecord.style.foreground;
+                } else {
+                    this._textPaint.color = paintRecord.style.color;
```

Comment:
any potential problem here? in the first branch,  this._textPaint is replaced with a new paint while in the second branch only the color of the current this._textPaint is modified. What will happen when paintRecord.style.foreground != null in the first build and  it then changes to null in the second build ?

Reply:
When `paintRecord.style.foreground != null`, `_textPaint` wiil be replaced by customized `ShaderPaint` , otherwise  it will use default `Paint _textPaint = new Paint { filterMode = FilterMode.Bilinear };` as `_textPaint` and only modify the `color` field when it comes in.